File: app/src/main/java/com/thales/github/data/GitHubDataSource.java
package com.thales.github.data;

import com.thales.github.model.pullrequest.PullRequest;
import com.thales.github.model.repository.Repositories;

import java.util.List;

import rx.Observable;

public interface GitHubDataSource {
    Observable<Repositories> getRepositories(String language, String sort, int page);

    Observable<List<PullRequest>> getPullRequests(String user, String repository);
}

File: app/src/main/java/com/thales/github/data/GitHubDataSource.kt
package com.thales.github.data

import com.thales.github.model.pullrequest.PullRequest
import com.thales.github.model.repository.Repositories

import rx.Observable

interface GitHubDataSource {
    fun getRepositories(language: String, sort: String, page: Int): Observable<Repositories>

    fun getPullRequests(user: String, repository: String): Observable<List<PullRequest>>
}
